### DISCUSSION TITLE
Is there a way to have one of frontmatter values be in pageTree?

### DISCUSSION THREAD

**Discussion by kyrylolvov** [fuma-nama/fumadocs#498] [Q&A] [answered]
I am implementing a custom layout and would like to ask if it is possible to have pageTree include a value from frontmatter.

Basically so I would specify request type in meta of the page, and then based on that put a specific icon, as right now I do it based on the first word of title.

<img width="251" alt="Screenshot 2024-06-20 at 11 11 11 AM" src="https://github.com/fuma-nama/fumadocs/assets/70720713/79811250-700b-4276-a997-524bf6158dc8">

**Comment by fuma-nama** [comment, marked as answer]
This can be done with:
https://fumadocs.vercel.app/docs/headless/source-api#attach-custom-properties

You can log the file data and see what properties it has, it should contain the frontmatter info.

**Comment by kyrylolvov** [comment]
Will leave this here, maybe someone will get into same problem, you also need to update schema of frontmatter for source

```
export const { getPage, getPages, pageTree } = loader({
  baseUrl: "/",
  rootDir: "",
  source: createMDXSource(map, {
    schema: {
      frontmatter: defaultSchemas.frontmatter.extend({
        requestType: z.string().optional(),
      }),
    },
  }),
  pageTree: {
    attachFile(node, file) {
      const { requestType } = file?.data.data as any;

      if (!requestType) return node;

      return { ...node, requestType };
    },
  },
});
```

**Reply by jpmti2016** [reply]
From where do you import `defaultSchemas`?

**Reply by fuma-nama** [reply]
This is for outdated versions of Fumadocs, now you can define frontmatter with collections schema field, please refer the docs
